core/tests/mod.rs

```diff
@@ -0,0 +1,778 @@
+        wsv,
+        queue: Arc::clone(&queue),
+        kura,
+        genesis_network: GenesisNetwork::test(configuration.clone(), true),
+        block_hashes: &vec![],
+        post_procedure: Arc::new(move |packet, peer_id| {
+            /*
+                let post = iroha_p2p::Post {
+                    data: iroha_core::NetworkMessage::SumeragiPacket(Box::new(packet.into())),
+                    peer_id: peer_id.clone(),
+                };
+                network1.post(post);
+            */
+            match packet.message {
+                message::Message::BlockCreated(block) => {
+                    let vb : VersionedCommittedBlock = block.block.commit_unchecked().into();
+                    *pb_hash2.lock().unwrap() = Some(vb.hash());
+                },
+                _ => (),
+            }
+        }),
+        update_topology_procedure: Arc::new(move |network_topology| {
+            /*
+                network2.update_topology(iroha_p2p::UpdateTopology(network_topology));
+            */
+        }),
+        get_connected_peers_procedure: Arc::new(move || {
+            /*
+                network3.online_peers(std::collections::HashSet::clone)
+            */
+            HashSet::new()
+        }),
+    });
+    
+    for _ in 0..300 {
+        if sumeragi.wsv(WorldStateView::clone).block_hashes_after_hash(None).len() == 1 {
+            if let Some(hash) = pb_hash.lock().unwrap().clone() {
+                assert_eq!(hash, sumeragi.wsv(WorldStateView::clone).block_hashes_after_hash(None)[0]);
+                return;
+            }
+        }
+        sleep_ms(100);
+    }
+    panic!("Sumeragi did not submit genesis.");
+}
+
+#[test]
+fn sumeragi_all_peers_commit_genesis() {
+    let mut configuration_for_logger = {
+        let mut sample_proxy =
+            iroha::samples::get_config_proxy(HashSet::new(), Some(KeyPair::generate().unwrap()));
+        let env_proxy = ConfigurationProxy::from_env();
+        
+        sample_proxy.override_with(env_proxy)
+                    .build()
+                    .expect("Test Iroha config failed to build. This is either a programmer error or a compiler bug.")
+    };
+    
+    let telemetry =
+        iroha_logger::init(&configuration_for_logger.logger).expect("Failed to initialize telemetry");
+        
+    let keypairs : Vec<KeyPair> = (0..7).into_iter().map(|n| KeyPair::generate().unwrap()).collect();
+    
+    let peer_ids : Vec<PeerId> = keypairs.iter().enumerate().map(|(i, kp)| PeerId::new(&format!("localhost:{}", i).parse().unwrap(), kp.public_key())).collect();
+
+    let mut sumeragis : HashMap::<PeerId, SumeragiHandle> = HashMap::new();
+    let mut configurations : HashMap::<PeerId, Configuration> = HashMap::new();
+    
+    for (index, (keypair, peer_id)) in keypairs.iter().zip(peer_ids.iter()).enumerate() {
+
+    let mut configuration = iroha::samples::get_config_proxy(peer_ids.iter().cloned().collect(), Some(keypair.clone())).build().unwrap();
```

Comment:
It's hard to read without spaces in loops

Reply:
Formatting. @SamHSmith if you're using my latest emacs, `apheleia` mode handles the formatting on save.

Reply:
Not any more, switched to https://github.com/focus-editor/focus

Will format before I undraft.